includes/REST/StoreSettingController.php

```diff
@@ -41,10 +41,48 @@ public function register_routes() {
 					'callback'            => [ $this, 'get_settings' ],
 					'permission_callback' => [ $this, 'get_settings_permission_callback' ],
 				],
+                [
+                    'methods'             => WP_REST_Server::EDITABLE,
+                    'callback'            => [ $this, 'update_settings' ],
+                    'permission_callback' => [ $this, 'get_settings_permission_callback' ],
+                ],
 			]
         );
     }
 
+    /**
+     * Update Store
+     *
+     * @param \WP_REST_Request $request
+     *
+     * @since DOKAN_LITE_SINCE
+     *
+     * @return WP_Error|\WP_REST_Response
+     */
+    public function update_settings( $request ) {
+        $vendor = $this->get_vendor();
+
+        if ( empty( $vendor->get_id() ) ) {
```

Comment:
kindly move this file into permission check block, also rewrite entire file accordingly (get_settings() methods also needs this checking)

Reply:
Resolved

Reply:
you can safely remove this checking from here, we don't need this checking anymore